### ISSUE TITLE
Fixed hostPort in Docker container does not work

### ISSUE THREAD

**Issue by jlarriba** [mesosphere/marathon#3272]
Using Marathon v0.15.2.

 If we select a fixed port as hostPort in the portMappings section of a Docker container, the container is never deployed. It keeps in "Waiting" state forever. However, if the hostPort is set to "0" (aleatory port), it works fine.

This json works as a charm:

```
{
  "id": "tomcat",
  "args": ["catalina.sh", "run"],
  "cpus": 0.5,
  "mem": 256.0,
  "container": {
    "type": "DOCKER",
    "docker": {
      "image": "tomcat",
      "network": "BRIDGE",
      "portMappings": [
        { "containerPort": 8080, "hostPort": 0 }
      ]
    }
  },
  "healthChecks": [
    {
      "protocol": "HTTP",
      "portIndex": 0,
      "path": "/",
      "gracePeriodSeconds": 5,
      "intervalSeconds": 20,
      "maxConsecutiveFailures": 3
    }
  ]
}
```

And this json does not deploy the docker container. It keeps it in "Waiting" state forever:

```
{
  "id": "tomcat",
  "args": ["catalina.sh", "run"],
  "cpus": 0.5,
  "mem": 256.0,
  "container": {
    "type": "DOCKER",
    "docker": {
      "image": "tomcat",
      "network": "BRIDGE",
      "portMappings": [
        { "containerPort": 8080, "hostPort": 32080 }
      ]
    }
  },
  "healthChecks": [
    {
      "protocol": "HTTP",
      "portIndex": 0,
      "path": "/",
      "gracePeriodSeconds": 5,
      "intervalSeconds": 20,
      "maxConsecutiveFailures": 3
    }
  ]
}
```

**Comment by philipnrmn** [comment]
Hi @jlarriba,

Are you sure that you have a node in your cluster in which this host port is available? If not, Marathon will simply wait for offers forever. 

You can check this by examining the Mesos offers in the Marathon logs.

**Comment by jlarriba** [comment]
Thank you very much @philipnrmn.

Even when I was sure that the port was available in all servers of the cluster, I dug into Marathon logs, I can see that when I try to deploy the container with a fixed port, Mesos answers with:

`Insufficient resources for [/tomcat] (need cpus=0.5, mem=256.0, disk=0.0, ports=([32080] required), available in offer: [id { value: "d415aa07-f535-40fa-a0ab-1469700a9b65-O20" } framework_id { value: "94d66dd8-6a18-4559-82ef-60ef6ff1cbc1-0000" } slave_id { value: "8fe421a4-9496-43ee-a472-d3d4de72ead4-S0" } hostname: "dns.hostname.com" resources { name: "cpus" type: SCALAR scalar { value: 2.0 } role: "*" } resources { name: "mem" type: SCALAR scalar { value: 2928.0 } role: "*" } resources { name: "disk" type: SCALAR scalar { value: 24980.0 } role: "*" } resources { name: "ports" type: RANGES ranges { range { begin: 31000 end: 32000 } } role: "*" } url { scheme: "http" address { hostname: "dns.hostname.com" ip: "xxx.xxx.xxx.xxx" port: 5051 } path: "/slave(1)" }]`

The point is that the cluster has the necessary resources, but the RANGES in the "ports" section got my eye, as it is setting a range between 31000 and 32000.

Indeed, when I start the container with a fixed hostPort of 31890, it starts fine.

Problem solved, now I need to figure out how to change the port range when Mesos has been installed via apt-get and started as a service.

Thank you very much.

**Comment by philipnrmn** [comment]
Thanks for the update, and I'm glad we were able to help!
